plugins/c8y_configuration_plugin/src/config.rs

```diff
@@ -1,14 +1,40 @@
 use c8y_smartrest::topic::C8yTopic;
 use mqtt_channel::Message;
 use serde::Deserialize;
+use std::collections::HashSet;
 use std::fs;
+use std::hash::{Hash, Hasher};
 use std::path::PathBuf;
 use tracing::{info, warn};
 
-#[derive(Deserialize, Debug, PartialEq, Default)]
+#[derive(Deserialize, Debug, Eq, PartialEq, Default)]
 #[serde(deny_unknown_fields)]
 pub struct PluginConfig {
-    pub files: Vec<String>,
+    pub files: HashSet<FileEntry>,
+}
+
+#[derive(Deserialize, Debug, Eq, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct FileEntry {
```

Comment:
I'm assuming that this struct was created to support future additions like `user`, `access-mode` etc.

Reply:
Exactly! This change makes us easier to support back-wards compatibility. The change from normal list (the previous format) to new format (mapping table) would be a breaking change for users.